ksqldb-engine/src/main/java/io/confluent/ksql/ddl/commands/CommandFactories.java

```diff
@@ -114,11 +116,14 @@ public DdlCommand create(
   }
 
   @Override
-  public DdlCommand create(final KsqlStructuredDataOutputNode outputNode) {
+  public DdlCommand create(
+      final KsqlStructuredDataOutputNode outputNode,
+      final Optional<RefinementInfo> emitStrategy
```

Comment:
Should this be passed as part of the output node, rather than separately? I could see the argument for this going either way, just tossing it out for consideration.

Reply:
I don't think it has anything to do with the output node? The old `suppress()` is just one more operator in the tree (and for the new emit final it's not even a new operator but a config to `aggregate()`.

`OutputNode` only contains a potential `LIMIT` clause and the result schema. It seem independent from the emit strategy?

Overall, passing in `emitStrategy` here is also not clean, but given how the code and planner is setup, it's the easiest way to do it this way. In a proper planner, we would do it differently, and make it a property of the logical aggregation node itself.

Reply:
That's fair, let's leave it out then. 

(FWIW, KsqlStructuredDataOutputNode contains a number of extra fields beyond those in OutputNode. It's still fair to argue that emitStrategy doesn't belong, though.)